retworkx-core/src/connectivity/conn_components.rs

```diff
@@ -0,0 +1,232 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use hashbrown::HashSet;
+use std::collections::VecDeque;
+use std::hash::Hash;
+
+use petgraph::visit::{GraphProp, IntoNeighborsDirected, IntoNodeIdentifiers, VisitMap, Visitable};
+use petgraph::{Incoming, Outgoing};
+
+/// Given an graph, a node in the graph, and a visit_map,
+/// return the set of nodes connected to the given node.
```

Comment:
I'd probably also mention here that this is in breadth first order and treating all edges as undirected.

Reply:
Done in https://github.com/Qiskit/retworkx/pull/595/commits/d5e5def3556d2221478c59cd334accb372573739.